### PR TITLE
chore: add support for mcp coverage MQ-2496

### PR DESCRIPTION
**Pull request by blaisedias**
- build coverage images
  - mayastor
  - mcp
- build coverage install bundles
- remove coverage files on cluster prior to running the install test
- retrieve coverage files on cluster prior to running the install test
- copy profile files to artifacts/coverage/data
- extract mayastor binaries from images used to generate coverage data  into artifacts/binaries/<tag>

### CODE REVIEW

File: scripts/remoteCoverageFiles.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    return results_dir
+
+
+def clearCoverageFiles(**kwargs):
+    '''
+    Clear mayastor coverage files from nodes
+    Relies on ssh working without password
+
+    '''
+    identity_file = kwargs['identity_file']
```
Comment (chriswldenyer) [inline scripts/remoteCoverageFiles.py:82]: Nit: getting the node addresses could be a separate function also called by getCoverageFiles()

**Review by chriswldenyer — APPROVED** [review]
LGTM, one comment

File: scripts/coverage-report.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  --archive             path to jenkins run archive
+  --tag                 tag of the build (required with --archive)
+  --data_dir            path to coverage dir, contains .profraw files
+  --binaries_dir        path to mayastor binaries
+  --report_dir          path to generate the coverage report
+  --mayastor            path to mayastor sources
+  --control-plane       path to mayastor control plane sources
+
+Either --archive or --data_dir and --binaries_dir can be specified,
+but no both sets of options.
```
Comment (chriswldenyer) [inline scripts/coverage-report.sh:20]: nit: no -> not

File: scripts/remoteCoverageFiles.py (hunk 1 above)
Comment (blaisedias) [reply scripts/remoteCoverageFiles.py]: Done.

File: scripts/coverage-report.sh (hunk 2 above)
Comment (blaisedias) [reply scripts/coverage-report.sh]: Done

File: pipelines/common/common.groovy [hunk 3]
```diff
@@ -577,5 +581,45 @@ def PopulateTestQueue(Map params) {
   }
 }
 
+def StashMayastorBinaries(Map params) {
+    def artefacts_stash_queue = params['artefacts_stash_queue']
+    def test_tag = params['e2e_image_tag']
+    def bin_dir = "./artifacts/binaries/${test_tag}"
+
+    sh "./scripts/getMayastorBinaries.py --tag ${test_tag} --registry ${env.REGISTRY} --outputdir ${bin_dir}"
+    stash_name = 'arts-bin'
```
Comment (rohan2794) [inline pipelines/common/common.groovy:590]:
```suggestion
    def stash_name = 'arts-bin'
```

**Review by rohan2794 — COMMENTED** [review]
minor comment

File: pipelines/common/common.groovy (hunk 3 above)
Comment (blaisedias) [reply pipelines/common/common.groovy]: fixed https://github.com/mayadata-io/mayastor-e2e/pull/323/commits/bd1c29368fa5ce761835f11e907ca24c59220a5c

**Review by rohan2794 — APPROVED** [review]
LGTM

File: src/common/locations/util_locations.go [hunk 4]
```diff
@@ -18,6 +18,17 @@ func locationExists(path string) string {
 	return path
 }
 
+// GetBuildInfoFile returns the path to build_info.json is one exists.
```
Comment (chriswldenyer) [inline src/common/locations/util_locations.go:21]: nit: "if" one exists

File: src/common/k8sinstall/util.go [hunk 5]
```diff
@@ -24,6 +24,37 @@ const InstallSuiteNameV1 = "Basic Install Suite (mayastor control plane)"
 const UninstallSuiteName = "Basic Teardown Suite"
 const UninstallSuiteNameV1 = "Basic Teardown Suite (mayastor control plane)"
 
+// postFixInstallation post fix installation yaml files for coverage and debug if required
+func postFixInstallation(yamlsDir string) error {
+	postfixOptions := ""
+	_, err := locations.GetBuildInfoFile()
```
Comment (chriswldenyer) [inline src/common/k8sinstall/util.go:30]: why not just put bfPath instead of the underscore here and remove line 32, or am I missing something?

**Review by chriswldenyer — APPROVED** [review]
LGTM (AFAICT with the Python)

File: src/common/locations/util_locations.go (hunk 4 above)
Comment (blaisedias) [reply src/common/locations/util_locations.go]: https://github.com/mayadata-io/mayastor-e2e/pull/323/commits/9df2eb195449ba4828a4a4ce9a47904402d1d7a3

File: src/common/k8sinstall/util.go (hunk 5 above)
Comment (blaisedias) [reply src/common/k8sinstall/util.go]: fixed https://github.com/mayadata-io/mayastor-e2e/pull/323/commits/9df2eb195449ba4828a4a4ce9a47904402d1d7a3

**Comment by blaisedias** [comment]
bors merge

**Comment by blaisedias** [comment]
bors merge

**Comment by blaisedias** [comment]
bors merge

**Comment by blaisedias** [comment]
bors merge

**Pull request merged**